client/hwlib/src/collectors/hardware_info.rs

```diff
@@ -0,0 +1,324 @@
+    let processor = devices::Processor {
+        codename: String::new(),
+        frequency: super::cpuinfo::read_max_cpu_frequency(max_cpu_frequency_filepath)?,
+        manufacturer: cpu_info.cpu_type,
+        version: cpu_info.model,
+    };
+    Ok(processor)
+}
+
+pub fn collect_chassis_info(
+    smbios_data: &smbioslib::SMBiosData,
+) -> Result<crate::models::devices::Chassis> {
+    let chassis_info = &smbios_data.collect::<smbioslib::SMBiosSystemChassisInformation>()[0];
+
+    let chassis_type = chassis_info.chassis_type().ok_or("").unwrap().to_string();
+    let manufacturer = chassis_info.manufacturer().to_string();
+    let sku = chassis_info.sku_number().to_string();
+    let version = chassis_info.version().to_string();
+
+    let chassis = crate::models::devices::Chassis {
+        chassis_type,
+        manufacturer,
+        sku,
+        version,
+    };
+
+    Ok(chassis)
+}
+
+pub fn collect_motherboard_info(smbios_data: &smbioslib::SMBiosData) -> Result<devices::Board> {
+    let board_info = &smbios_data.collect::<smbioslib::SMBiosBaseboardInformation>()[0];
+
+    let manufacturer = board_info.manufacturer().to_string();
+    let product_name = board_info.product().to_string();
+    let version = board_info.version().to_string();
+
+    let board = devices::Board {
+        manufacturer,
+        product_name,
+        version,
+    };
+
+    Ok(board)
+}
+
+pub fn collect_motherboard_info_from_device_tree(
+    device_tree_filepath: &'static str,
+) -> Result<devices::Board> {
+    let base_path = std::path::Path::new(device_tree_filepath);
+
+    let manufacturer = std::fs::read_to_string(base_path.join("model"))
+        .map(|s| s.trim().to_string())
+        .unwrap_or_else(|_| "Unknown".to_string());
+
+    let product_name = std::fs::read_to_string(base_path.join("compatible"))
+        .map(|s| s.trim().to_string())
+        .unwrap_or_else(|_| "Unknown".to_string());
+
+    let version = std::fs::read_to_string(base_path.join("model"))
+        .map(|s| s.trim().to_string())
+        .unwrap_or_else(|_| "Unknown".to_string());
+
+    let board = devices::Board {
+        manufacturer,
+        product_name,
+        version,
+    };
+
+    Ok(board)
+}
+
+pub fn get_system_info(smbios_data: &smbioslib::SMBiosData) -> Result<(String, String)> {
+    let system_info = &smbios_data.collect::<smbioslib::SMBiosSystemInformation>()[0];
+    Ok((
+        system_info.product_name().to_string(),
+        system_info.manufacturer().to_string(),
+    ))
+}
+
+#[cfg(target_os = "linux")]
+/// Overwrite smbioslib::table_load_from_device to use parameters for files
+/// so we can test our code without reading smbios data from the machine that
+/// runs the test
+pub(crate) fn table_load_from_device(
+    entry_file: &'static str,
+    table_file: &'static str,
+) -> Result<smbioslib::SMBiosData, std::io::Error> {
+    let version: smbioslib::SMBiosVersion;
```

Comment:
Generally, `let` declarations with no value should be avoided as much as reasonably possible. In this case, I think the fix is quite straightforward so this ought to be removed (see suggestion below)